pinot-common/src/main/java/org/apache/pinot/common/function/scalar/regexp/RegexpLikeConstFunctions.java

```diff
@@ -34,17 +35,54 @@ public class RegexpLikeConstFunctions {
   @ScalarFunction
   public boolean regexpLike(String inputStr, String regexPatternStr) {
     if (_matcher == null) {
-      _matcher = PatternFactory.compile(regexPatternStr).matcher("");
+      Pattern p = PatternFactory.compile(regexPatternStr);
+      _matcher = p.matcher("");
     }
 
     return _matcher.reset(inputStr).find();
   }
 
+  @ScalarFunction
+  public boolean regexpLike(String inputStr, String regexPatternStr, String matchParameter) {
+    if (_matcher == null) {
+      Pattern p = buildPattern(regexPatternStr, matchParameter);
+      _matcher = p.matcher("");
+    }
+
+    return _matcher.reset(inputStr).find();
+  }
+
+  private Pattern buildPattern(String pattern, String matchParameter) {
+    // Validate that all characters in matchParameter are supported
+    for (char c : matchParameter.toCharArray()) {
```

Comment:
Given we only allow `'i'` or `'c'` as the match parameter, consider simplifying the handling to:
- Check single character
- Check character is `'i'` or `'c'` or throw exception